test/dag.spec.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const IPFSApi = require('../src')
+const f = require('./utils/factory')
+
+describe('.dag', function () {
+  this.timeout(50 * 1000) // slow CI
+
+  let ipfs
+  let ipfsd
+
+  const obj = {
+    a: 1,
+    b: [1, 2, 3],
+    c: {
+      ca: [5, 6, 7],
+      cb: 'foo'
+    }
+  }
+
+  before((done) => {
+    f.spawn({ initOptions: { bits: 1024 } }, (err, _ipfsd) => {
+      expect(err).to.not.exist()
+      ipfsd = _ipfsd
+      ipfs = IPFSApi(_ipfsd.apiAddr)
+      done()
+    })
+  })
+
+  after((done) => {
+    if (!ipfsd) return done()
+    ipfsd.stop(done)
+  })
+
+  it('.dag.tree', (done) => {
+    const expectedPaths = [
+      'a', 'b', 'b/0', 'b/1', 'b/2', 'c', 'c/ca',
+      'c/ca/0', 'c/ca/1', 'c/ca/2', 'c/cb'
+    ]
+
+    ipfs.dag.put(obj, (err, cid) => {
+      expect(err).to.not.exist()
+      ipfs.dag.tree(cid, {}, (err, paths) => {
+        expect(err).to.not.exist()
+        expect(paths).deep.equal(expectedPaths)
+        done()
+      })
+    })
+  })
+
+  it('.dag.put', (done) => {
```

Comment:
Looks like a left-over from copy&pasting. Please remove it if that's the case.

Reply:
I did not know how to name this test. I looked into `log.spec.js` and `repo.spec.js` and tried to follow the convention there.

Reply:
I was just wondering as this test doesn't have a call to `tree()`.